modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DomainDeploymentTestCase.java

```diff
@@ -1941,6 +1945,74 @@ public void pluginDeclaredInDomainIsAbleToLoadClassesExportedByTheAppWhereItIsUs
     withContextClassLoader(anotherClassLoader, () -> completionCallback.error(new NullPointerException()));
   }
 
+  @Test
+  @Issue("MULE-19376")
+  public void pluginDeclaredInDomainIsAbleToGetResourceWithSameNameInAppAndExtensionFromExtension() throws Exception {
+    // Given a plugin which loads classes.
+    final ArtifactPluginFileBuilder pluginWhichLoadsClasses = loadClassExtensionPlugin;
+    // Given a plugin which creates a connection.
+    final ArtifactPluginFileBuilder pluginWhichCreatesConnection = classloaderConnectExtensionPlugin;
+
+    // Given a domain depending on the plugin.
+    DomainFileBuilder domainFileBuilder = new DomainFileBuilder("domain-with-test-plugin")
+        .definedBy("empty-domain-config.xml")
+        .dependingOn(pluginWhichLoadsClasses);
+
+    // Given an app depending on the domain and exporting a class.
+    final ApplicationFileBuilder applicationFileBuilder =
+        new ApplicationFileBuilder("app-with-connection").definedBy("app-with-connection.xml")
+            .containingClass(echoTestClassFile, "org/foo/EchoTest.class")
+            .configuredWith(EXPORTED_PACKAGES, "org.foo")
+            .dependingOn(domainFileBuilder)
+            .dependingOn(pluginWhichCreatesConnection)
+            .containingResource("file.txt", "file.txt");
+
+    addPackedDomainFromBuilder(domainFileBuilder);
+    addPackedAppFromBuilder(applicationFileBuilder);
+
+    startDeployment();
+
+    assertDeploymentSuccess(domainDeploymentListener, domainFileBuilder.getId());
+    assertDeploymentSuccess(applicationDeploymentListener, applicationFileBuilder.getId());
+
+    String fileMessage = executeApplicationFlowAndReturnMessage("flowWhichConnects");
+    assertThat(fileMessage, is(equalTo("extension file")));
```

Comment:
instead of using a literal, read the file content and set in constant so that you don't have problems in case you change the content.